File: api/schemas/stats.py
from pydantic import BaseModel
from typing import Optional

class StatsSummary(BaseModel):
    total_distance: float
    avg_fuel_efficiency: Optional[float]
    total_consumed_fuel: Optional[float]
    total_records: int

class MonthlyStats(BaseModel):
    year: int
    month: int
    total_distance: float
    avg_fuel_efficiency: Optional[float]
    total_consumed_fuel: Optional[float]
    record_count: int

class StatsResponse(BaseModel):
    summary: StatsSummary
    monthly: list[MonthlyStats]

File: frontend/src/types/stats.ts
export interface StatsSummary {
  total_distance: number;
  avg_fuel_efficiency: number | null;
  total_consumed_fuel: number | null;
  total_records: number;
}

export interface MonthlyStats {
  year: number;
  month: number;
  total_distance: number;
  avg_fuel_efficiency: number | null;
  total_consumed_fuel: number | null;
  record_count: number;
}

export interface StatsResponse {
  summary: StatsSummary;
  monthly: MonthlyStats[];
}
